wic/iot2050-swu.wks.in

```diff
@@ -0,0 +1,20 @@
+#
+# Copyright (c) Siemens AG, 2021
+#
+# Authors:
+#  Quirin Gylstorff <quirin.gylstorff@siemens.com>
+#
+# This file is subject to the terms and conditions of the MIT License.  See
+# COPYING.MIT file in the top-level directory.
+#
+
+bootloader --ptable gpt --append "console=ttyS3,115200n8 earlycon=ns16550a,mmio32,0x02810000 mtdparts=47040000.spi.0:512k(ospi.tiboot3),2m(ospi.tispl),4m(ospi.u-boot),128k(ospi.env),128k(ospi.env.backup),1m(ospi.sysfw),64k(pru0-fw),64k(pru1-fw),64k(rtu0-fw),64k(rtu1-fw),-@8m(ospi.rootfs) rootwait ${EXTRA_KERNEL_PARAMS}"
+
+# rootfs 1
+part --source rootfs-u-boot --sourceparams="no_initrd=yes,script_prepend=env exists sysselect || setenv sysselect 1,root=PARTLABEL=rootfs\"\$\{sysselect\}\"" --fixed-size 4G --fstype ext4 --label rootfs1 --align 1024
+
+# rootfs 2
+part --source rootfs-u-boot --sourceparams="no_initrd=yes,script_prepend=env exists sysselect || setenv sysselect 1,root=PARTLABEL=rootfs\"\$\{sysselect\}\"" --fixed-size 4G --fstype ext4 --label rootfs2 --align 1024
```

Comment:
Do we have to fill both partitions? The initial boot image would shrink if we could make the second partition a pure placeholder.

Reply:
The partition doesn't need to be filled. The partition needs to have the correct size before applying the partition update.

Reply:
I could add resizing during first boot.

Reply:
Resize of the data partition? I though that would take place already?

Reply:
I mean resizing both partition `rootfs2` and `data`.

Reply:
I don't think you need to touch rootfs2, just create that partition without filling it. data can be placed after it, formatted and usable, and it will continue to be auto-expanded on first boot.

Reply:
The only problem is when you update your system with SWUpdate and afterwards you flash an image with an empty rootfs2 the boot process will fail.

Reply:
OK, because sysselect will not be reset when only reflashing the mass storage - understood...